PyPortal_Azure_Plant_Monitor/code.py

```diff
@@ -52,15 +77,27 @@
         temperature = ss.get_temp()
         # display soil sensor values on pyportal
         gfx.display_moisture(moisture_level)
-        temperature = gfx.display_temp(temperature)
+        gfx.display_temp(temperature)
+
         print('Sending data to Azure')
         gfx.display_azure_status('Sending data...')
-        hub.send_device_message(temperature)
-        hub.send_device_message(moisture_level)
+
+        # send the temperature and moisture level to Azure
+        message = {
+            "Temperature": temperature,
+            "MoistureLevel": moisture_level
+        }
+        device.send_telemetry(json.dumps(message))
+        device.loop()
+
         gfx.display_azure_status('Data sent!')
         print('Data sent!')
     except (ValueError, RuntimeError) as e:
         print("Failed to get data, retrying\n", e)
         wifi.reset()
+        wifi.connect()
```

Comment:
Q: Does `wifi.reset()` not handle reconnections? We may want to add a `reconnect` method to ESP32SPI's `wifimanager` if it doesn't

Reply:
It doesn't reconnect automatically - I tested this out to make sure.

Raised https://github.com/adafruit/Adafruit_CircuitPython_ESP32SPI/issues/100 to add this, will try to implement it later today if I get a chance.

Reply:
@brentru - adding to this. The WiFi manager doesn't reconnect, but all its methods will connect if it's not connected, meaning that after a `reset`, a call to `get` will work.

When using the socket like I am in the Azure library, the connection is not remade.

Add more comments on this to the issue mentioned above. Would be good to get your thoughts.